addons/pos_sale/models/account_move.py

```diff
@@ -0,0 +1,24 @@
+from odoo import models, _
+
+
+class AccountMove(models.Model):
+    _inherit = 'account.move'
+
+    def reflect_cancelled_sol(self, isCancelled):
+        if self.env.user.has_group('point_of_sale.group_pos_user'):
+            for invoice in self:
+                for pos_order_line in invoice.pos_order_ids.mapped('lines'):
+                    if isCancelled and "(Cancelled)" not in pos_order_line.sale_order_line_id.name:
+                        pos_order_line.sale_order_line_id.name = _("%(line_description)s (Cancelled)", line_description=pos_order_line.sale_order_line_id.name)
+                    elif not isCancelled:
+                        pos_order_line.sale_order_line_id.name = pos_order_line.sale_order_line_id.name.replace(" (Cancelled)", "")
+
+    def button_cancel(self):
+        res = super().button_cancel()
+        self.reflect_cancelled_sol(True)
+        return res
```

Comment:
Can you tell me when this is needed ?

Reply:
This is used when we have invoiced that down payment and then cancel that invoice it should be reflected the same way, I have attached the ref video and same for the action confirm.
[Screencast from 05-12-24 03:54:50 PM IST.webm](https://github.com/user-attachments/assets/00fc6335-ec3c-474f-b004-adf2c95f66cd)

Reply:
Mmh that's weird since the pos_order itself still invoiced and not cancelled ?

Reply:
No, we have invoiced the pos_order linked with that SO, than we cancelled that invoice that was reflected on the SOL (as that downpayment was because of that pos_order), which was not reflected before.